packages/grafana-ui/src/components/uPlot/plugins/TooltipPlugin.tsx

```diff
@@ -99,57 +102,63 @@ export const TooltipPlugin: React.FC<TooltipPluginProps> = ({
     const tooltipInterpolator = config.getTooltipInterpolator();
 
     if (tooltipInterpolator) {
-      // Custom toolitp positioning
+      // Custom tooltip positioning
       config.addHook('setCursor', (u) => {
-        tooltipInterpolator(
-          setFocusedSeriesIdx,
-          setFocusedPointIdx,
-          (clear) => {
-            if (clear) {
-              setCoords(null);
-              return;
-            }
-
-            if (!bbox) {
-              return;
-            }
-
-            const { x, y } = positionTooltip(u, bbox);
-            if (x !== undefined && y !== undefined) {
-              setCoords({ x, y });
-            }
-          },
-          u
-        );
-      });
-    } else {
-      config.addHook('setLegend', (u) => {
-        if (!isMounted()) {
+        if (!bbox || !isMounted()) {
           return;
         }
-        setFocusedPointIdx(u.legend.idx!);
-        setFocusedPointIdxs(u.legend.idxs!.slice());
-      });
 
+        ReactDOM.unstable_batchedUpdates(() => {
```

Comment:
Wouldn't having all the state values in one object i.e. similar as to how state in class based components is stored on `this.state`, or maybe even using `useReducer` have the same effect?

Reply:
maybe, might require more changes. tooltipInterpolator calls these setters internally:

https://github.com/grafana/grafana/blob/b89f8fdfcbf70dc3a2c3a7208d12fea87f7979a0/public/app/plugins/panel/barchart/bars.ts#L271-L309

not sure what route is preferred here. @dprokop ?

Reply:
I would prefer single `setState` rather than unstable methods. @leeoniya can you try if this will help? If so, we can then consider tooltip interpolator API update to accept a single state setter rather than multiple fns.